client/src/components/DataEntry/index.tsx

```diff
@@ -150,14 +148,17 @@ const DataEntry: React.FC = () => {
   }, [electionId, auditBoard])
 
   const submitMembers = async (members: IAuditBoardMember[]) => {
-    await putMembers(
+    const response1 = await putMembers(
       electionId,
       auditBoard!.jurisdictionId,
       auditBoard!.roundId,
       auditBoardId,
       members
     )
-    setAuditBoard(await loadAuditBoard())
+    if (!response1) return
+    const response2 = await loadAuditBoard()
+    if (!response2) return
+    setAuditBoard(response2)
```

Comment:
this pattern seems a little tedious. before, the rejected promises short-circuited the function wherever there was a failure. now we have to manually do it ourselves. seems like we lose some of the zen of promises. I can't think of any way around it except letting the exception bubble up and trying to catch it at a higher level in the app (maybe with a React error boundary?)

alternatively we could just not short circuit on failure

Reply:
@jonahkagan True, it did seem rather tedious. There are quite a few places where we handle an api failure in different ways, so it was kinda handy there. Trapping the error inside the api call for the toast is generally cleaner, but I also like the idea of using error boundaries. :thinking:

Reply:
I've never used error boundaries before, so can't really speak to whether they would be a good fit here. Since you already have this working, I'd say we just stick with it